### PR TITLE
SBT plugin best practices

### PR DESCRIPTION
**Pull request by JosephEarl**
<http://www.scala-sbt.org/0.13/docs/Plugins-Best-Practices.html>

- Use autoImport, namespace all auto-imported keys and types
- Separate Checkstyle from CheckstylePlugin to allow easier integration with other plugins
- Add list of all settings to README
- Rename the config sbt-tests so they all start with checkstyle-config-

### CODE REVIEW

File: src/sbt-test/checkstyle/checkstyle-auto/build.sbt [hunk 1]
```diff
@@ -1,11 +1,9 @@
-version := "0.1"
+version := "2.1.1-SNAPSHOT"
```
Comment (ajsquared) [inline src/sbt-test/checkstyle/checkstyle-auto/build.sbt]: Are they recommending changing the version in all the scripted tests to match the version of the plugin itself?

**Comment by ajsquared** [comment]
Thanks for doing this, looks good!  There was one scalastyle error though: https://gist.github.com/ajsquared/167b7da37206df7b13de

**Comment by JosephEarl** [comment]
Will fix that, thanks

File: src/sbt-test/checkstyle/checkstyle-auto/build.sbt (hunk 1 above)
Comment (JosephEarl) [inline src/sbt-test/checkstyle/checkstyle-auto/build.sbt]: They're not, can revert that

**Comment by ajsquared** [comment]
Thanks again!  I'll get a new version published with these change shortly.

**Pull request merged**